src/main/java/hoodini/Deadline.java

```diff
@@ -0,0 +1,54 @@
+package hoodini;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Handles deadline inputs from the user
+ * and stores the deadline date.
+ */
+public class Deadline extends Input {
+    private LocalDate deadline;
+
+    /**
+     * Stores the deadline input by the user
+     * @param input User input, parses user input and stores the deadline.
+     * @throws DateTimeParseException If the date is not in the correct format.
+     */
+    public Deadline(String input) throws DateTimeParseException {
+        super(input.split(" ", 2)[1]
+                .split("/", 2)[0]);
+        this.deadline = LocalDate.parse(input
+                .split("/by")[1]
+                .trim());
+
+    }
```

Comment:
I've noticed that some of your JavaDoc comments do not have @return sections. Maybe you could include one for each JavaDoc comment to let people know what it returns?